MongoDB.ruleset

```diff
@@ -4,27 +4,61 @@
     <Rule Id="UseConfigureAwait" Action="Warning" />
   </Rules>
   <Rules AnalyzerId="Microsoft.Analyzers.ManagedCodeAnalysis" RuleNamespace="Microsoft.Rules.Managed">
```

Comment:
I gather there is no way to share these rules amongst all projects?

Reply:
I agree that it'd be great if we could share a single ruleset among all the projects. We currently have 3: Legacy, LegacyTests, and "normal" (i.e.` MongoDB.rulset`) that share a lot of rules, but have subtle variations among the three. Perhaps as future improvement we can use a single ruleset? I'll create a ticket to track this.

Reply:
The reason there are 4 rulesets is that they are all configured differently, some more strictly than others (MongoDB.ruleset is the strictest and the test and Legacy rulesets relax things in various ways).

Eventually we can have just one ruleset (equal to or perhaps even stricter than MongoDB.ruleset), but that will require making many code changes to comply with the stricter rules.